src/Pos/Explorer/Web/Server.hs

```diff
@@ -277,40 +287,51 @@ getBlocksPage mPageNumber pageSize = do
     -- Get total pages from the blocks.
     totalPages <- getBlocksPagesTotal pageSize
 
-    -- Initially set on the last page number.
-    let pageNumber      = fromMaybe totalPages $ toInteger <$> mPageNumber
-    let calculateOffset = pageNumber * pageSizeInt
-    let pageNumberInt   = toInteger pageNumber
-
-    -- Get total blocks in the blockchain.
-    blocksTotal <- toInteger <$> getBlocksTotal
+    -- Initially set on the last page number if page number not defined.
+    let pageNumber = fromMaybe totalPages $ toInteger <$> mPageNumber
 
     -- Make sure the parameters are valid.
-    when (pageNumberInt > totalPages) $
+    when (pageNumber <= 0) $
+        throwM $ Internal "Number of pages must be greater than 0."
+
+    when (pageNumber > totalPages) $
         throwM $ Internal "Number of pages exceeds total pages number."
 
+    -- TODO: Fix in the future.
+    when (pageSize /= 10) $
```

Comment:
This will be removed, but generally, I agree, we should use constants from a single source.